etc/config/metadata/component-registries/elyra-kfp-preconfig.json

```diff
@@ -0,0 +1,11 @@
+{
+  "display_name": "KFP Preloaded Components",
+  "metadata": {
+    "description": "Preloaded components that are supported by Kubeflow Pipelines",
+    "runtime": "kfp",
+    "location_type": "Directory",
+    "categories": ["Preloaded KFP"],
+    "paths": ["kfp"]
```

Comment:
Yeah this gets into that sticky situation where we load the preconfigured filename-based components from `share/jupyter/metadata/components` (they are moved there from `etc/config/components` on build). So the filename-based readers (and all subclasses) check first to see if the entered path is an absolute path, and if not, assumes the location to be at `os.path.join(ENV_JUPYTER_PATH[0], 'components', path)`. If the resulting path is still inaccurate, the file or directory is skipped and logged as such.

Reply:
If we wanted to display this in such a way that it might be less confusing for the user, we could potentially add some logic to (what is currently the) SchemaFilter where it just replaces the relative path with the absolute path

Reply:
That won't work because this is actually in the _instance_ of the schema, not the schema itself.  However, schemas can reference `metadata_class_name` values that essentially derive from [`Metadata`](https://github.com/elyra-ai/elyra/blob/7b286ecbcf9b709391da3693798fe2e2e8162cc9/elyra/metadata/metadata.py#L31) that have `post_load` and `pre_save` method overrides, so such a class _could_ look for these kinds of things - validate their location, and update the "relative location".  The metadata-test schemas [show an example](https://github.com/elyra-ai/elyra/blob/7b286ecbcf9b709391da3693798fe2e2e8162cc9/elyra/metadata/schemas/metadata-test.json#L7) if we want to entertain this.

Reply:
Oh right! Not sure what I was thinking there, I think I was confusing it with the `post_load` method itself. Anyway, yeah maybe just another thing to keep in mind for a follow up